### ISSUE TITLE
Question about ensuring message delivery between reconnects

### ISSUE THREAD

**Issue by yangm97** [emqx/emqtt#149]
I see emqtt once had an automatic reconnect feature but is has been removed. However I haven't found any recommended replacement.

Doing a quick test reveals unexpected disconnects (for instance, tcp_closed) results in a exit from the emqtt client process, which propagates to my process.

I could probably trap that exit signal and issue a reconnect from my proccess but then what about messages in flight?

Suppose I want to ensure QoS 1-2 messages got sent, do I need to build a queue within my process state and pop items from that queue only after receiving a PUBACK?

**Comment by qzhuyan** [comment]
@yangm97 I have added reconnects in this PR with option `{reconnect, true}`
https://github.com/emqx/emqtt/pull/143

Have you tried that ?

**Comment by yangm97** [comment]
@qzhuyan nope, if I remember correctly I used the latest release available at that time.

**Comment by qzhuyan** [comment]
Any updates?
